### PR TITLE
Respect autogenerate setting for JS projects

### LINKED ISSUE #18124: Typescript type autogeneration config is ignored in 4.13.7
### Required System information

- Node.js version: 18.17.0
- NPM version: 9.6.7
- Strapi version: 4.13.7
- Database: postgres
- Operating system: macOS
- Is your project Javascript or Typescript: Typescript

### Describe the bug

After setting the typescript config as below, types are generated and it breaks the typescript compilation (https://github.com/strapi/strapi/issues/18106)

```ts
// config/typescript.ts
export default () => ({
    autogenerate: false,
});
```

### Steps to reproduce the behavior

1. Upgrade to 4.13.7
2. Install and build
3. run `npm run dev`

### Expected behavior

Types are not generated.

### LINKED ISSUE #19195: Typescript type autogeneration is done for non typescript projects
### Required System information

- Node.js version: 18.14.2
- NPM version: 9.5.1
- Strapi version: 4.17.0
- Database: sqlite3
- Operating system: macOS 14.2.1
- Is your project Javascript or Typescript: Javascript

_Did you know you can run `yarn strapi report` or `npm run strapi report` to get this information easier?_

No, but it does not tell if you are using TS or JS.

### Describe the bug

The typescript types will be auto generated under `types/generated` .

### Steps to reproduce the behavior

```sh
npx create-strapi-app@4.17.0 --quickstart --no-run new-strapi-project
cd new-strapi-project

# note that npx create-strapi-app uses yarn to install your deps! Does not matter for the next step.

npm run develop
```

At this point, stop the server and if you look in the root of the `new-strapi-project` you can see a folder `types`.

If you create a file to disable the autogeneration as described on https://docs.strapi.io/dev-docs/configurations/typescript#strapi-specific-configuration-for-typescript that does not make any difference.

### Expected behavior

- No typescript types generated for javascript projects
- If you have `autogenerate` set to `false` the types should not be generated!

### Code snippets

If applicable, add code samples to help explain your problem.

### Additional context

This issue is similar: #18124 
The issue should probably be tracked in: #15225

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

The --develop command in a JS Strapi project now skips the type generation step when config/typescript.js autogenerate = false

### Why is it needed?

Types are not a hard requirement for JS projects and it's annoying in many cases to have it update after every change.

### How to test it?

In a javascript Strapi project with a config/typescript.js containing `module.exports = { autogenerate: false }`, go to the content type builder and make some change, and see the server does not generate types

In all other project configurations, it should still generate types.

### Related issue(s)/PR(s)

Fixes #18124 (autogeneration is required on TS projects for the develop command or the server will not start)
Fixes #19195

### CODE REVIEW

File: packages/core/admin/_internal/node/develop.ts [hunk 1]
```diff
@@ -230,20 +230,27 @@ const develop = async ({
...
-      strapi: strapiInstance,
-      pwd: cwd,
-      rootDir: undefined,
-      logger: { silent: true, debug: false },
-      artifacts: { contentTypes: true, components: true },
-    });
+    // if this is a JS project (no tsconfig) AND they have disabled type autogeneration, skip type generation
+    if (!tsconfig?.config && strapi.config.get('typescript.autogenerate') === false) {
+      logger.log('Skipping typescript autogeneration');
+    }
```
Comment (joshuaellis) [inline packages/core/admin/_internal/node/develop.ts:236]: Is this way round correct? i.e. you couldn't turn it off if you're in a TS project right? But you _might_ want too? and you have to set the config to turn it off in a JS project which seems like extra steps?

Also do we need to log this? If so I would use `info` not `log` & if not we can just use `debug`?

**Review by Convly — APPROVED** [review]
As mentionned already I'm not a fan of allowing not generating types, but let's add this experimental option back for now.

Only enabling this for JS project is a good compromise.

Also, is there any documentation to update related to this? (I don't remember if we document this feature only for JS or any JS/TS projects)

**Comment by innerdvations** [comment]
@pwizla Not sure how you want to handle this, but I'll give you a quick run-down.

This doc: https://docs.strapi.io/dev-docs/configurations/typescript should have been updated a few months ago to remove the experimental autogenerate flag.

However, if this PR is merged, it would need to be kept in, but updated to mention it only works for javascipt projects because autogeneration is required for typescript projects.

**Comment by pwizla** [comment]
Thanks, @innerdvations. I drafted [this PR](https://github.com/strapi/documentation/pull/1975), is this what you meant? Feel free to review and edit the text directly :)

In the meantime, shall I also create another shorter PR just to remove the "experimental" callout?

**Review by Convly — APPROVED** [review]

**Pull request merged**